site/src/pages/DeploySettingsPage/ServiceBannerSettingsPage.tsx

```diff
@@ -0,0 +1,245 @@
+import TextField from "@material-ui/core/TextField"
+import { useActor } from "@xstate/react"
+import { FeatureNames } from "api/types"
+import {
+  Badges,
+  DisabledBadge,
+  EnterpriseBadge,
+  EntitledBadge,
+} from "components/DeploySettingsLayout/Badges"
+import { Header } from "components/DeploySettingsLayout/Header"
+import { LoadingButton } from "components/LoadingButton/LoadingButton"
+import { Stack } from "components/Stack/Stack"
+import { FormikContextType, useFormik } from "formik"
+import React, { useContext, useState } from "react"
+import { Helmet } from "react-helmet-async"
+import { pageTitle } from "util/page"
+import * as Yup from "yup"
+import { XServiceContext } from "xServices/StateContext"
+import { getFormHelpers } from "util/formUtils"
+import makeStyles from "@material-ui/core/styles/makeStyles"
+import FormControlLabel from "@material-ui/core/FormControlLabel"
+import Switch from "@material-ui/core/Switch"
+import { BlockPicker } from "react-color"
+import { useTheme } from "@material-ui/core/styles"
+import FormHelperText from "@material-ui/core/FormHelperText"
+import Button from "@material-ui/core/Button"
+
+export const Language = {
+  messageLabel: "Message",
+  backgroundColorLabel: "Background Color",
+  updateBanner: "Update",
+  previewBanner: "Preview",
+}
```

Comment:
Sure. What do you by "gets stale without warning"?

Reply:
The translation aspect I've never understood, because it's unlikely we'd translate the product for years.

Reply:
@ammario, the Language objects are just POJOs so no warning is thrown if we cease to use a string in our JSX. Thus you can have key-value pairs that aren't used, and this was happening more frequently as we grew the code base and refactored.

Regarding translations in general: I don't think we need translations, but I didn't introduce them; we had them when I arrived in the form of these Language objects. I just figured if we were going to bother with translating at all, we might as well use a very popular and robust solution versus something home-baked that wasn't maintainable.

There [are a couple of other reasons](https://www.notion.so/coderhq/2022-07-22-28f0f1432c394f4b962853e7063993ef#43bfac3e2e8741f99d92e6fd99717cc2) we might want to use i18next: it's really nice to have strings all colocated in one area so that we can make sure the verbiage we use is consistent. Libraries like i18next come with features that handle quirks like pluralization and interpolation - thus we don't have to have giant if-blocks in our JSX. Lastly, although the day may be far in the future, if we ever want to translate it's nice to start now when our code base is relatively small and simple.

My two cents! Happy to discuss further online if you'd like. Also happy to consider just putting strings directly in the JSX. I'd just rather not use the Language objects.

Reply:
@ammario I was told the Language blocks were a way to get started on translations without committing to a library. They're actually kind of annoying to use in tests as if you pull in different components, you have to alias each Language block separately, but maybe there's a benefit I'm overlooking.

To add HTML or links, you'll want to use the library's [Trans](https://www.notion.so/coderhq/V2-Frontend-Code-Conventions-and-Best-Practices-024466624dfa482e894570e95a6cc842#c30e5cc0da9445b4bb3ef556f45bc18f) component. Mathias recently merged up a great example, [here](https://github.com/coder/coder/pull/4810/files#diff-5ba21dd0a2c0a1fa74e22d8b5b613f56f4940a22488acb78157e2a8ba4190044R40). I'll always agree that i18n is more of a hassle than straight JSX. If it feels particularly painful, we can certainly take it out, but should probably discuss in FE variety first; feel free to throw something on the agenda if you'd like!